Catalog/Examples/UI/ToggleButton/FollowButton.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  FollowButton.swift
+//  Catalog
+//
+//  Created by Domagoj Hustnjak on 4/4/20.
+//  Copyright © 2020 Infinum. All rights reserved.
+//
+
+import UIKit
+
+@IBDesignable final class FollowButton: UIButton {
+
+    // MARK: - Public properties -
+
+    var toggleState: ToggleState = .available(active: false)
+
+    // MARK: - Private properties -
+
+    private var activityIndicator: UIActivityIndicatorView!
+
+    // MARK: - Lifecycle -
+
+    override func awakeFromNib() {
+        configureActivityIndicator()
+    }
+}
+
+// MARK: - Extensions -
+
+extension FollowButton: Togglable {
+
+    func toggle(with toggleState: ToggleState) {
+        self.toggleState = toggleState
+
+        switch toggleState {
+        case .available(let active):
+            activityIndicator.stopAnimating()
+            let title = active ? "Following" : "Follow"
+            setTitle(title, for: .normal)
+            isHidden = false
+
+        case .unavailable:
+            isHidden = true
+
+        case .inProgress:
+            setTitle("", for: .normal)
+            activityIndicator.startAnimating()
+        }
+    }
+}
+
+private extension FollowButton {
+
+    func configureActivityIndicator() {
+        activityIndicator = UIActivityIndicatorView(style: .white)
+        let buttonHeight = bounds.size.height
+        let buttonWidth = bounds.size.width
+        activityIndicator.center = CGPoint(x: buttonWidth / 2, y: buttonHeight / 2)
```

Comment:
Couple of things here 🙂 

Firstly, I'd make this activityIndicator a lazy property, so instead the current `private var activityIndicator: UIActivityIndicatorView!` I'd do the following: 

```
    let activityIndicator = UIActivityIndicatorView(style: .white)
    addSubview(activityIndicator)
    return activityIndicator
```

Next thing - bounds can be unknown at start or they can be changed if the button gets resized for whatever reason. This should be moved to layout subviews, like so: 

```
   override func layoutSubviews() {
        let buttonHeight = bounds.size.height
        let buttonWidth = bounds.size.width
        activityIndicator.center = CGPoint(x: buttonWidth / 2, y: buttonHeight / 2)
    }
```

After doing those two things you actually don't need `configureActivityIndicator()` or `awakeFromNib()` anymore.